catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloadManager.java

```diff
@@ -85,6 +87,7 @@ public void cleanUp() {
    *     should read from
    * @throws DownloadException
    */
+  @Override
   public ResourceResponse download(
       ResourceRequest resourceRequest, Metacard metacard, ResourceRetriever retriever)
       throws DownloadException {
```

Comment:
Should this be updated to call `cache.put`? Should the cache logic be updated as well to use the put in addition to the `get` methods for cache management? Even if the impl isn't changed for this, might be nice to at least call the method and let the implementation handle whether it's needed or not.

Reply:
The cache is updated by the `ReliableResourceDownloader` which is started by this `ReliableResourceDownloadManager` method.  The manager does not wait for the download to fully complete before returning.  Currently we only call the cache when the resource is fully downloaded and ready to be cached.

https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloader.java#L330